### ISSUE TITLE
PROJEKT-2: Hämta mottagaruppgifter från ett moln-AD?

### ISSUE THREAD

**Issue by MonaAdawi** [studio27se/ehub#127]
**Beskriv den lösning du önskar**
Kan vi göra en utveckling att hämta mottagaruppgifter (Affv mottagare mail och mobilnr) från ett moln-AD till Ehub mottagarlistor? Vad behöver ni veta för att lämna ett estimat?

**Förslag eller förfrågan**
Är ditt ärende ett förslag för framtida förbättre, eller vill ni skapa en förfrågan om ändring? Vid förfrågan svarar vi alltid med ett estimat, när alla detaljer retts ut, innan arbetet påbörjas.

**Ytterligare sammanhang**
Lägg till annan kontext eller skärmbilder om funktionen här.

**Comment by FredrikElliot** [comment]
Jag tror vi behöver veta mer kring syftet. Försök svara på varför det behövs, så kan vi nog få mer förståelse kring vilka lösningar som kan passa för att lösa problemet ni ser.
Direkta frågor att tänka på utöver detta är också: Ska alla de kunna logga in och ändra sina inställningar själva? Vad ska hända om en person slutar? Ska det finnas default-inställningar som appliceras på alla som börjar automstiskt? 
@MonaAdawi @johulls

**Comment by MonaAdawi** [comment]
@FredrikElliot vi har bokat internt möte för att diskutera och återkomma med kravspec till er (v.40)

**Comment by FredrikElliot** [comment]
Möte med Affärsverken Intern IT, 2024-09-25

IT-mottagare = bara anställda på Affv, Karlskrona Hem, Kruthusen
Hålla en lista uppdaerad för att nå användare via sms/mail.
Om mailservern funkar är det fullgott att skicka mail. Kan man justera detta med hjälp av kritikalitet?
Saknar idag uppdaterad sms-lista. I Adt finns både email och telefonnummer. Avhängigt.

Inläggningsbehov:
Fritextsmeddelande som ska kunna styra till mail eller sms (egen orsakstext?).
Val om det ska publiceras på webben eller inte. (Här behöver vi testa hur detta skulle se ut på webben)
Rör bara affärsverken? eller rör det även systerbolagen? Ska detta kanske vara valbara objekt?

**Comment by MonaAdawi** [comment]
@FredrikElliot du får gärna tagga denna med: Projekt

**Comment by FredrikElliot** [comment]
Nu har vi tagit fram estimat på båda lösningsförslagen, enligt diskussion på möte med projektgrupp och IT. Lösningarna är kortfattat:
1. Integration mot Affärsverkens AD (MS Entra) via en registrering av Enterprise Application och synk av användare i en specifik grupp.
2. Nedladdning och import av csv-fil från SFTP-lagring vid exakt tidpunkt.

Båda förslagen estimeras ta ca 12 timmar, exklusive test. Vi rekommenderar förslag 1; Framförallt av säkerhetsskäl.

@monaadawi @johulls

**Comment by MonaAdawi** [comment]
@FredrikElliot vi väljer alt 1 och ni kan starta den så snart ni kan. Tack.
@johulls

**Comment by MonaAdawi** [comment]
@FredrikElliot kan du återkomma med en status på denna så jag vet vart vi står och när vi kan förvänta oss att denna blir helt klar?

**Comment by FredrikElliot** [comment]
@monaadawi Yes! Denna ligger med i vår nästa sprint som sträcker sig från denna veckan hela vägen till 9/1 där tanken är att det ska implementeras på UAT för test.

**Comment by MonaAdawi** [comment]
@FredrikElliot härligt tack!!

**Comment by FredrikElliot** [comment]
Denna är nu redo för test. Dock behöver vi synka lite hur/när vi gör det. 
@monaadawi

**Comment by FredrikElliot** [comment]
@monaadawi Vi kan börjar med en synk. Men framförallt handlar det om två tester:
- Vi börjar med att prova att synka in alla användare manuellt (och kollar att ni får in det ni vill/behöver). 
- Vi provar att synka enligt schemat (04.00) så att allt funkar.

Varför det är viktigt att vi gör detta kontrollerat är ju att ni kommer få in en hel massa personer som mottagare i er testmiljö. Ni vill såklart inte att dessa ska bli nersmsade med era teststörningar :)

DOCK, är det ju möjligt att skapa en AD-grupp för "personer som ska vara mottagare i testmiljön" och använda den gruppen i AD-synken i UAT. Det är upp till er.
